File: Stage20/C/Problem15651.c
#include <stdio.h>

int N, M;
int arr[9];

void dfs(int depth) {
    if (depth == M) {
        for (int i = 0; i < M; i++) printf("%d ", arr[i]);
        printf("\n");
        return;
    }

    for (int i = 1; i <= N; i++) {
        arr[depth] = i;
        dfs(depth + 1);
    }
}

int main(void) {
    scanf("%d %d", &N, &M);

    dfs(0);
    return 0;
}

File: Stage20/Python/Problem15651.py
def dfs(depth):
    if depth == M:
        print(*arr)
        return

    for i in range(1, N + 1):
        arr.append(i)
        dfs(depth + 1)
        arr.pop()

def main():
    global N, M, arr
    N, M = map(int, input().split())
    arr = []
    dfs(0)
    
if __name__ == "__main__":
    main()